lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -191,11 +270,35 @@ class QcFlagService {
                 runNumber,
                 dplDetectorId,
             });
+            await this._resetQcFlagEffectivePeriod(newInstance);
 
             const createdFlag = await QcFlagRepository.findOne(this.prepareQueryBuilder().where('id').is(newInstance.id));
 
             await createdFlag.addDataPasses(targetRun.dataPass);
 
+            const flagsToBePartiallyDiscarded = await QcFlagRepository.findAll({
```

Comment:
I would rather name it `intersectingFlags`

Reply:
refactored